openmls/src/framing/mls_auth_content.rs

```diff
@@ -0,0 +1,430 @@
+//! # MLS content authentication
+//!
+//! This module contains structs and implementation that pertain to content
+//! authentication in MLS. Besides structs that directly represent structs in
+//! the MLS specification, this module also contains
+//! [`VerifiableAuthenticatedContent`], a wrapper struct which ensures that the
+//! signatures are verified before the content of an MLS [`PrivateMessage`] or
+//! [`PublicMessage`] can be accessed by processing functions of OpenMLS.
+use crate::{
+    ciphersuite::signable::{Signable, SignedStruct, Verifiable, VerifiedStruct},
+    error::LibraryError,
+    group::errors::ValidationError,
+};
+
+#[cfg(doc)]
+use super::{PrivateMessage, PublicMessage};
+
+use super::{
+    mls_content::{ContentType, FramedContentBody, FramedContentTbs},
+    AddProposal, Commit, ConfirmationTag, Credential, CredentialBundle, FramingParameters,
+    GroupContext, GroupEpoch, GroupId, Proposal, Sender, Signature, WireFormat,
+};
+use openmls_traits::OpenMlsCryptoProvider;
+use std::io::{Read, Write};
+
+use serde::{Deserialize, Serialize};
+use tls_codec::{
+    Deserialize as TlsDeserializeTrait, Serialize as TlsSerializeTrait, Size, TlsSerialize, TlsSize,
+};
+
+/// Private module to ensure protection of [`AuthenticatedContent`].
+mod private_mod {
+    #[derive(Default)]
+    pub(crate) struct Seal;
+}
+
+/// 7.1 Content Authentication
+///
+/// ```c
+/// // draft-ietf-mls-protocol-16
```

Comment:
```suggestion
```

Reply:
Since draft-17 is out you could reference that